HW2/P5/driver.py

```diff
@@ -80,3 +97,24 @@ def randcolor():
             # SUBPROBLEM 3: sort objects by location.  Be sure to update the
             # grid if objects' indices change!  Also be sure to sort the
             # velocities with their object positions!
+
+            # Get the order of locations based on Morton ordering
+            zorder = [tomorton(int(positions[i,0] * grid_size), int(positions[i,1] * grid_size)) for i in range(num_balls)]
+            sorted_positions = np.argsort(zorder)
+
+            # 1. update each ball's position according to its Morton order
+            positions = positions[sorted_positions]
+
+            # 2. update each ball's velocities according to its Morton order
+            velocities = velocities[sorted_positions]
+
+            # 3.update grid indices
+            for i in range(num_balls):
+                # Boundary check
+                if positions[i,0]>=0 and positions[i,1]>=0 and positions[i,0]<=1 and positions[i,1]<=1:
+                    positions_coords = (positions[i,] / grid_spacing).astype(np.int)
+                    grid[positions_coords[0], positions_coords[1]] = i
+
+
```

Comment:
Good work on variable definition and code management, everything is very readable.